### ISSUE TITLE
Type definitions missing in eslint-plugin-jsdoc v53+, plugin imported as any

### ISSUE THREAD

**Issue by yunarch** [gajus/eslint-plugin-jsdoc#1449]
## Expected behavior

In previous version **52.0.4**, `eslint-plugin-jsdoc` exported complete TypeScript type definitions for the plugin, including `ConfigGroups`, `ConfigVariants`, `ErrorLevelVariants`, and a strongly-typed `configs` object.
Importing the plugin provided accurate type information instead of falling back to `any`.

Example from v52.0.4:

```ts
// index.d.ts
export type ConfigGroups = "recommended" | "stylistic" | "contents" | "logical" | "requirements";
export type ConfigVariants = "" | "-typescript" | "-typescript-flavor";
export type ErrorLevelVariants = "" | "-error";

/**
 * @type {import('eslint').ESLint.Plugin & {
 *   configs: Record<`flat/${ConfigGroups}${ConfigVariants}${ErrorLevelVariants}`, import('eslint').Linter.Config>
 * }}
 */
declare const index: import("eslint").ESLint.Plugin & {
  configs: Record<`flat/${ConfigGroups}${ConfigVariants}${ErrorLevelVariants}`, import("eslint").Linter.Config>;
};

export default index;
```

This allowed full IntelliSense and type safety.

## Actual behavior

In **v53+**, the type definitions were reduced to:

```ts
// index.d.ts
import plugin_default from "./plugin.js";
export { plugin_default as default };
```

This removes the `ConfigGroups`, `ConfigVariants`, `ErrorLevelVariants`, and `configs` typings, causing the plugin to be imported as `any`, breaking type safety and editor IntelliSense.

## Reproducible example

```ts
import jsdoc from "eslint-plugin-jsdoc";

jsdoc.nonExistentProperty; // no TS error but should produce an error as for v54.0.4
```

You should see an error such as:
> Property 'nonExistentProperty' does not exist on type 'Plugin & { configs: Record<"flat/recommended" | "flat/recommended-error" | "flat/recommended-typescript" | "flat/recommended-typescript-error" | "flat/recommended-typescript-flavor" | ... 24 more ... | "flat/requirements-typescript-flavor-error", Config<...>>; }'.ts(2339)

Ts config if need it:
```json
{
"compilerOptions": {
    "target": "ESNext",
    "module": "Preserve",
    "composite": false,
    "skipLibCheck": true,
    "moduleResolution": "Bundler",
    "isolatedModules": true,
    "inlineSources": false,
    "esModuleInterop": true,
    "declarationMap": true,
    "noEmit": true,
    "allowImportingTsExtensions": true,
    "strict": true,
    "strictNullChecks": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "forceConsistentCasingInFileNames": true,
    "declaration": true,
    "preserveWatchOutput": true
  },
  "exclude": ["node_modules"]
}
```

**Comment by lishaduck** [comment]
I think the issue is tsdown is exporting from -`.js` files, which works fine in `dist/`, so attw doesn't catch it, but errors when it's in node_modules.
As a quick bandage, I think you can set `maxNodeModuleJsDepth` to 1 (or two or something idk).

We could either revert v53 as v54, then once it's fixed release v55, or just see how quickly we get a fix.
If we deprecated v53, npm won't resolve it, but I think pnpm & yarn still prefer `latest` over "latest engines-compatible non-deprecated" like modern npm, so that's sort of an option?
